tests/unit/test_spack.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Test minimal snap."""
+
+import logging
+import os
+import pathlib
+import re
+import shlex
+import subprocess
+import unittest
+
+logger = logging.getLogger(__name__)
+
+
+class TestSnap(unittest.TestCase):
+    """Unit test Spack snap."""
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        """Test class setup."""
+        subprocess.run(shlex.split("tox -e clean"))
+        logger.info("Building snap")
+        # Go up 3 levels from test file
+        os.chdir("/".join(__file__.split("/")[:-3]))
+        subprocess.run(shlex.split("tox -e snap"))
+        # Find generated snap
+        cls.SPACK = re.findall(r"(spack.*?snap)", " ".join(os.listdir()))[0]
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        """Test class teardown."""
+        subprocess.run(shlex.split("tox -e clean"))
```

Comment:
I usually place `tearDownClass(cls) -> None` at the end of the test case class just so I know what is coming after all my tests finish running. I find it makes tests written with `unittest` more readable. Just a stylistic preference of mine :sweat_smile:

Reply:
Makes sense, I don't have a preference so I'll update it.